base/src/Module.cpp

```diff
@@ -1034,6 +1034,17 @@ bool Module::relay(boost::shared_ptr<Module> next, bool open)
 	return queueCommand(cmd);
 }
 
+void Module::flushQue()
+{
+	// recursively call the flushQue of children module
+	for (auto it = mModules.begin(); it != mModules.end(); ++it)
+	{
+		it->second->flushQue();
+	}
+	LOG_INFO << "flushQue for <" << myId << ">";
+	mQue->flush();
```

Comment:
wouldnt it be better to move "mQue->flush()" to be the first line of this function ?

Reply:
Also it is better to call this function as flushQueRecursive() for clarity.
and have another function called flushQue() which just does mQue->flush();
that will be more useful

Reply:
There is also this wierd possibilty with multiple threads that frames which are under process will still get forwarded since they wont be flushed :(

Reply:
Is flushQue() method really required ? We have getQue() which gives the queue and we can call flush on it.

Reply:
I will add it anyway